apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -196,7 +197,54 @@ export default class SchoolInfoInterstitial extends React.Component {
     };
   }
 
+  isBlank(field) {
+    // return true when field is blank
+    return !!(!field || field.trim() === '');
+  }
+
+  validateSubmission = () => {
+    const {country, schoolType, schoolName, ncesSchoolId} = this.state;
+    let errors = {};
+    let isValid = true;
+
+    if (this.isBlank(country)) {
+      errors.country = true;
+      isValid = false;
+    } else {
+      if (this.isBlank(schoolType)) {
+        errors.schoolType = true;
+        isValid = false;
+      } else {
+        if (country === 'United States') {
+          const ncesSchoolType = ['public', 'private', 'charter'];
+          if (ncesSchoolType.includes(schoolType)) {
+            if (this.isBlank(ncesSchoolId)) {
+              errors.ncesSchoolId = true;
+              isValid = false;
+            }
+
+            if (ncesSchoolId === '-1' && this.isBlank(schoolName)) {
```

Comment:
I'm still about confused about where "-1" comes from? If the ncesSchoolId is -1 when the school id doesn't exist, do we ever get into the branch above?

Reply:
ncesSchoolId is set to '-1' when the checkbox (I cannot find my school above) is clicked.  A school id is returned when a school is selected from the drop-down.  A school name (including other attributes that are filled out) is returned when a user manually inputs a school name.  We do enter this branch when the ^ condition is met (condition;  US, NCES, non-dropdown, manual input)

Reply:
For clarity, I'm asking if we ever get into the branch on like 221 (sorry if that was unclear). Is there ever a case where ncesSchoolId is blank? It sounds like it's either a school id or it's -1?

Reply:
Yes, it's possible try to submit a blank `ncesSchoolId`:

- It will be blank when the school dropdown is not displayed (in which case it's not required) or when it is displayed but the user hasn't interacted with it yet (neither selected a school _nor_ indicated that they can't find their school.)
- It will be `-1` when the user has checked the "Can't find my school" checkbox, or selected "Another school not listed here" in the dropdown.

Reply:
Thanks for the clarification!